.github/scripts/cabal-gild.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+"""
```

Comment:
Could we replace this whole script by:

```bash
#!/usr/bin/env bash
set -euf -o pipefail

git ls-files *.cabal cabal.project \
  | grep -E -v '^clash-vexriscv/' \
  | xargs -I {} cabal-gild -i {} -o {}
```

?

Reply:
Perhaps there is a way to make `git ls-files` ignore subdirectories, but I couldn't get it working...

Reply:
That would format all files in `git ls-files`, but we want to check it they are formatted on CI.
I did try simply changing the `cabal-gild` invocation to check the files instead of formatting them.

However, since it simply exits with exitcode 1, the only feedback you get if one of the files is not formatted is `input is not formatted`

Reply:
Yup that worked, here's an example: https://github.com/bittide/bittide-hardware/actions/runs/10160062525